node.js

```diff
@@ -71,129 +92,171 @@ function isInModuloRange(input_value, lower_bound, include_lower, upper_bound, i
-  callback(null, _self);
+    console.log("vvvvv     vvvvv     Summary     vvvvv     vvvvv");
+    console.log("FingerTable: \n", FingerTable);
+    console.log("Predecessor: ", predecessor);
+    console.log("^^^^^     ^^^^^     End Summary     ^^^^^     ^^^^^")
+    callback(null, _self);
 }
 
 function fetch({ request: { id } }, callback) {
-  console.log(`Requested User ${id}`);
-  if (!users[id]) {
-    callback({ code: 5 }, null); // NOT_FOUND error
-  } else {
-    callback(null, users[id]);
-  }
+    console.log(`Requested User ${id}`);
+    if (!users[id]) {
+        callback({ code: 5 }, null); // NOT_FOUND error
+    } else {
+        callback(null, users[id]);
+    }
 }
 
 function insert({ request: user }, callback) {
-  if (users[user.id]) {
-    const message = `Err: ${user.id} already exits`;
-    console.log(message);
-    callback({ code: 6, message }, null); // ALREADY_EXISTS error
-  } else {
-    users[user.id] = user;
-    const message = `Inserted User ${user.id}:`;
-    console.log(message);
-    callback({ status: 0, message }, null);
-  }
+    if (users[user.id]) {
+        const message = `Err: ${user.id} already exits`;
+        console.log(message);
+        callback({ code: 6, message }, null); // ALREADY_EXISTS error
+    } else {
+        users[user.id] = user;
+        const message = `Inserted User ${user.id}:`;
+        console.log(message);
+        callback({ status: 0, message }, null);
+    }
 }
 
-/* added 20191102 */
+/**
+ * Directly implement the pseudocode's find_successor() method.
+ * 
+ * However, it is able to discern whether to do a local lookup or an RPC.
+ * If the querying node is the same as the queried node, it will stay local.
+ * 
+ * @argument id value being searched
+ * @argument node_querying node initiating the query
+ * @argument node_queried node being queried for the ID
+ * @returns id.successor
+ * 
+ * @version 20191103
+ */
 async function find_successor(id, node_querying, node_queried) {
-    /**
-     * Directly implement the pseudocode's find_successor() method.
-     * However, it is able to discern whether to do a local lookup or an RPC.
-     * If the querying node is the same as the queried node, it will stay local.
-     */
     // enable debugging output
     const DEBUGGING_LOCAL = false;
 
-    let n_prime_successor;
+    let n_prime = NULL_NODE;
+    let n_prime_successor = NULL_NODE;
 
-    /* vvvvv     vvvvv     TBD debugging code     vvvvv     vvvvv */
     if (DEBUGGING_LOCAL) {
         console.log("vvvvv     vvvvv     find_successor     vvvvv     vvvvv");
     }
-    /* ^^^^^     ^^^^^     TBD debugging code     ^^^^^     ^^^^^ */
+
     if (node_querying.id == node_queried.id) {
         // use local value
         // n' = find_predecessor(id);
-        let n_prime = await find_predecessor(id);
+        try {
+            n_prime = await find_predecessor(id);
+        } catch (err) {
+            n_prime = NULL_NODE;
```

Comment:
Out of curiosity, why are you choosing to not console log the errors here? I would think this would be the most useful form of logging.